src/block_cache/block_chunkcache.c

```diff
@@ -706,27 +778,33 @@ __wt_chunkcache_setup(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig
           &chunkcache->free_bitmap));
     }
 
-    WT_RET(__config_get_sorted_pinned_objects(session, cfg, &pinned_objects, &cnt));
-    chunkcache->pinned_objects = pinned_objects;
-    chunkcache->pinned_entries = cnt;
+    WT_ERR(__wt_rwlock_init(session, &chunkcache->pinned_objects.array_lock));
+    WT_ERR(__config_get_sorted_pinned_objects(session, cfg, &pinned_objects, &cnt));
+    __wt_writelock(session, &chunkcache->pinned_objects.array_lock);
```

Comment:
I did that for consistency, but since we are sure no other threads would be using the array, I will get rid of the locks. Updated in [8583581](https://github.com/wiredtiger/wiredtiger/pull/9466/commits/858358123fc3ee82cee5078abc678f38c00aad0e)